File: app/src/test/java/com/techventus/wikipedianews/util/ArrayUtilTest.java
package com.techventus.wikipedianews.util;

import org.junit.Test;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.HashSet;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class ArrayUtilTest {

    // Tests for isNullOrContainsEmpty(T[] array)

    @Test
    public void testIsNullOrContainsEmpty_nullArray() {
        assertTrue(ArrayUtil.isNullOrContainsEmpty(null));
    }

    @Test
    public void testIsNullOrContainsEmpty_emptyArray() {
        assertTrue(ArrayUtil.isNullOrContainsEmpty(new String[0]));
    }

    @Test
    public void testIsNullOrContainsEmpty_arrayWithNullElements() {
        assertTrue(ArrayUtil.isNullOrContainsEmpty(new String[]{null, "test"}));
        assertTrue(ArrayUtil.isNullOrContainsEmpty(new String[]{"test", null}));
        assertTrue(ArrayUtil.isNullOrContainsEmpty(new String[]{null, null}));
    }

    @Test
    public void testIsNullOrContainsEmpty_arrayWithNonNullElements() {
        assertFalse(ArrayUtil.isNullOrContainsEmpty(new String[]{"a", "b"}));
    }

    @Test
    public void testIsNullOrContainsEmpty_arrayWithMixOfNullAndNonNullElements() {
        assertTrue(ArrayUtil.isNullOrContainsEmpty(new String[]{"a", null, "c"}));
    }

    @Test
    public void testIsNullOrContainsEmpty_arrayWithEmptyString() {
        // Current implementation of isNullOrContainsEmpty does not check for empty strings,
        // it only checks for null objects. This test reflects that.
        // If the requirement was to check for empty strings as well, this test would fail
        // and the method would need to be updated.
        assertFalse(ArrayUtil.isNullOrContainsEmpty(new String[]{""}));
        assertFalse(ArrayUtil.isNullOrContainsEmpty(new String[]{"a", ""}));
    }


    // Tests for isNullOrEmpty(Collection<T> collection)

    @Test
    public void testIsNullOrEmpty_nullCollection() {
        assertTrue(ArrayUtil.isNullOrEmpty(null));
    }

    @Test
    public void testIsNullOrEmpty_emptyArrayList() {
        assertTrue(ArrayUtil.isNullOrEmpty(new ArrayList<String>()));
    }

    @Test
    public void testIsNullOrEmpty_emptyHashSet() {
        assertTrue(ArrayUtil.isNullOrEmpty(new HashSet<String>()));
    }

    @Test
    public void testIsNullOrEmpty_collectionWithElements() {
        assertFalse(ArrayUtil.isNullOrEmpty(Arrays.asList("a", "b")));
        assertFalse(ArrayUtil.isNullOrEmpty(new HashSet<>(Arrays.asList("a", "b"))));
    }

    @Test
    public void testIsNullOrEmpty_collectionWithNullElements() {
        // isNullOrEmpty only checks if the collection itself is null or empty,
        // not the contents of the collection.
        assertFalse(ArrayUtil.isNullOrEmpty(Arrays.asList(null, "a")));
        assertFalse(ArrayUtil.isNullOrEmpty(Collections.singletonList(null)));
    }
}

File: app/src/test/java/com/techventus/wikipedianews/util/ArrayUtilTest.kt
package com.techventus.wikipedianews.util

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import java.util.Arrays

class ArrayUtilTest {

    @Test
    fun testIsNullOrContainsEmpty_nullArray() {
        assertTrue(isNullOrContainsEmpty(null as Array<String?>?))
    }

    @Test
    fun testIsNullOrContainsEmpty_emptyArray() {
        assertTrue(isNullOrContainsEmpty(emptyArray<String?>()))
    }

    @Test
    fun testIsNullOrContainsEmpty_arrayWithNullElements() {
        assertTrue(isNullOrContainsEmpty(arrayOf<String?>(null, "test")))
        assertTrue(isNullOrContainsEmpty(arrayOf<String?>("test", null)))
        assertTrue(isNullOrContainsEmpty(arrayOf<String?>(null, null)))
    }

    @Test
    fun testIsNullOrContainsEmpty_arrayWithNonNullElements() {
        assertFalse(isNullOrContainsEmpty(arrayOf("a", "b")))
    }

    @Test
    fun testIsNullOrContainsEmpty_arrayWithMixOfNullAndNonNullElements() {
        assertTrue(isNullOrContainsEmpty(arrayOf<String?>("a", null, "c")))
    }

    @Test
    fun testIsNullOrContainsEmpty_arrayWithEmptyString() {
        // Current implementation of isNullOrContainsEmpty does not check for empty strings,
        // it only checks for null objects. This test reflects that.
        // If the requirement was to check for empty strings as well, this test would fail
        // and the method would need to be updated.
        assertFalse(isNullOrContainsEmpty(arrayOf("")))
        assertFalse(isNullOrContainsEmpty(arrayOf("a", "")))
    }


    // Tests for isNullOrEmpty(collection: Collection<T>?)

    @Test
    fun testIsNullOrEmpty_nullCollection() {
        assertTrue(isNullOrEmpty(null as Collection<String>?))
    }

    @Test
    fun testIsNullOrEmpty_emptyArrayList() {
        assertTrue(isNullOrEmpty(ArrayList<String>()))
    }

    @Test
    fun testIsNullOrEmpty_emptyHashSet() {
        assertTrue(isNullOrEmpty(HashSet<String>()))
    }

    @Test
    fun testIsNullOrEmpty_collectionWithElements() {
        assertFalse(isNullOrEmpty(Arrays.asList("a", "b")))
        assertFalse(isNullOrEmpty(HashSet(Arrays.asList("a", "b"))))
    }

    @Test
    fun testIsNullOrEmpty_collectionWithNullElements() {
        // isNullOrEmpty only checks if the collection itself is null or empty,
        // not the contents of the collection.
        assertFalse(isNullOrEmpty(Arrays.asList(null, "a")))
        assertFalse(isNullOrEmpty(java.util.Collections.singletonList(null as String?)))
    }
}
